includes/abstracts/abstract-wc-payment-token.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+/**
+ * WooCommerce Payment Token
+ *
+ * Representation of a general payment token to be extended by individuals types of tokens
+ * examples: Credit Card, eCheck.
+ *
+ * @class 		WC_Payment_Token
+ * @since		2.6.0
+ * @package		WooCommerce/Abstracts
+ * @category	Abstract Class
+ * @author		WooThemes
+ */
+ abstract class WC_Payment_Token {
+
+ 	/** @protected int Token ID */
+ 	protected $id;
+ 	/** @protected array Core Token Data (stored in the payment_tokens table) */
+ 	protected $data;
+ 	/** @protected array Meta Token Data (extra data associated with a payment token, stored in the payment_token_meta table) */
+ 	protected $meta;
+
+ 	/**
+	 * Initialize a payment token
+	 *
+	 * These fields are accepted by all payment tokens:
+	 * default      - boolean Optional - Indicates this is the default payment token for a user
+	 * type         - string  Required - WC core ships with 'cc' or 'echeck' but other values can be used for custom payment token types
+	 * token        - string  Required - The actual token to store
+	 * gateway_id   - string  Required - Identifier for the gateway this token is associated with
+	 * customer_id  - int     Optional - ID for the customer this token is associated with. 0 if this token is not associated with a user
+	 *
+	 * @since 2.6.0
+	 * @param string $id Token ID
+	 * @param array $data Core token data
+	 * @param array $meta Meta token data
+	 */
+	public function __construct( $id, $data, $meta ) {
+		$this->id = $id;
+		$this->data = $data;
+		$this->meta = $meta;
+	}
+
+	/**
+	 * Returns the payment token ID
+	 *
+	 * @since 2.6.0
+	 * @return string Token ID
+	 */
+	public function get_id() {
+		return $this->id;
+	}
+
+	/**
+	 * Returns the raw payment token
+	 *
+	 * @since 2.6.0
+	 * @return string Raw token
+	 */
+	public function get_token() {
+		return $this->data['token'];
+	}
+
+	/**
+	 * Set the raw payment token
+	 *
+	 * @since 2.6.0
+	 * @param string $token
+	 */
+	public function set_token( $token ) {
+		$this->data['token'] = $token;
+	}
+
+	/**
+	 * Returns the type of this payment token (cc, echeck, or something else)
+	 *
+	 * @since 2.6.0
+	 * @return string Payment token type
+	 */
+	public function get_type() {
+		return isset( $this->data['type'] ) ? $this->data['type'] : null;
+	}
+
+	/**
+	 * Set the payment token type
+	 *
+	 * @since 2.6.0
+	 * @param string $type
+	 */
+	public function set_type( $type ) {
+		$this->data['type'] = $type;
+	}
+
+	/**
+	 * Returns the customer ID associated with the token or false if this token is not associated
+	 *
+	 * @since 2.6.0
+	 * @return mixed Customer ID if this token is associated with a user or false.
+	 */
+	public function get_customer_id() {
```

Comment:
I'd prefer to see this and the `set` method (probably the DB column name too) changed  to `user_id`, since that's a bit more semantically accurate. This would match how the [`WC_Order` object refers to the user associated with an order](https://github.com/woothemes/woocommerce/blob/2.5.2/includes/abstracts/abstract-wc-order.php#L946-L954), and it removes any confusion around whether the customer ID refers to the WC order user ID or the gateway customer ID.